src/views/checkups/self-validation/utils/selfValidationPermissions.ts

```diff
@@ -0,0 +1,250 @@
+  }
+
+  // Create or update ClusterRoleBinding
+  try {
+    const existingClusterRoleBinding = await k8sGet({
+      model: ClusterRoleBindingModel,
+      name: SELF_VALIDATION_CLUSTER_ROLE_BINDING,
+    });
+
+    // Check if the ServiceAccount for this namespace is already in the subjects
+    const crb = existingClusterRoleBinding as IoK8sApiRbacV1ClusterRoleBinding;
+    const hasSubjectForNamespace = crb.subjects?.some(
+      (subject) =>
+        subject?.kind === ServiceAccountModel.kind &&
+        subject?.name === SELF_VALIDATION_SA &&
+        subject?.namespace === namespace,
+    );
+
+    if (!hasSubjectForNamespace) {
+      // Add the ServiceAccount for this namespace to the subjects
+      const updatedSubjects = [
+        ...(crb.subjects || []),
+        { kind: ServiceAccountModel.kind, name: SELF_VALIDATION_SA, namespace },
+      ];
+
+      await k8sPatch({
+        data: [{ op: 'replace', path: '/subjects', value: updatedSubjects }],
+        model: ClusterRoleBindingModel,
+        resource: existingClusterRoleBinding,
+      });
+    }
+  } catch (error) {
+    // If k8sGet fails, ClusterRoleBinding doesn't exist, try to create it
+    if (error?.code === 404 || error?.json?.code === 404) {
+      try {
+        await k8sCreate({
+          data: selfValidationClusterRoleBinding(namespace),
+          model: ClusterRoleBindingModel,
+        });
+      } catch (createError) {
+        if (createError?.code === 403 || createError?.json?.code === 403) {
+          throw new Error(getPermissionDeniedMessage('create'));
+        }
+        throw createError;
+      }
+    } else {
+      // Re-throw other errors (including permission errors from k8sPatch)
+      if (error?.code === 403 || error?.json?.code === 403) {
+        throw new Error(getPermissionDeniedMessage('update'));
+      }
+      throw error;
+    }
+  }
+};
+
+export const uninstallPermissions = async (namespace: string) => {
+  try {
+    const existingClusterRoleBinding = await k8sGet({
+      model: ClusterRoleBindingModel,
+      name: SELF_VALIDATION_CLUSTER_ROLE_BINDING,
+    });
+
+    // Remove the ServiceAccount for this namespace from the subjects
+    const crb = existingClusterRoleBinding as IoK8sApiRbacV1ClusterRoleBinding;
+    const updatedSubjects = (crb.subjects || []).filter(
+      (subject) =>
+        !(
+          subject?.kind === ServiceAccountModel.kind &&
+          subject?.name === SELF_VALIDATION_SA &&
+          subject?.namespace === namespace
+        ),
+    );
+
+    if (updatedSubjects.length === 0) {
+      // If no subjects left, delete the entire ClusterRoleBinding
+      await k8sDelete({
+        model: ClusterRoleBindingModel,
+        resource: { metadata: { name: SELF_VALIDATION_CLUSTER_ROLE_BINDING } },
+      });
+    } else {
+      // Update the ClusterRoleBinding with the remaining subjects
+      await k8sPatch({
```

Comment:
using `test` would improve reliability in the case of concurrent modifications. We might overwrite a server state either by deleting resource with non-empty subject list or by overwriting a list that changed in the meantime.